### PR TITLE
ENH: Confounds metadata

### LINKED ISSUE #1617: Write out metadata about Global Signal regressors
From #1301, they should contain the following description:

```
{
    "GlobalSignal": {"Method": "Mean"},
    "WhiteMatter": {"Method": "Mean"},
    "CSF": {"Method": "Mean"}
}
```

### LINKED ISSUE #1618: Write out metadata about ICA-AROMA regressors
From #1301:

``AROMAaggrCompXX`` should indicate ICA-AROMA version. Not sure whether ``"Method"`` can be a dict with ``"Name"`` and ``"Version"``, or what.

### PR DESCRIPTION
**Pull request by rciric**
* Resolves #1617 
* Resolves #1618

### CODE REVIEW

File: fmriprep/workflows/bold/confounds.py [hunk 1]
```diff
@@ -687,6 +690,13 @@ def _getusans_func(image, thresh):
     ica_aroma_confound_extraction = pe.Node(ICAConfounds(err_on_aroma_warn=err_on_aroma_warn),
                                             name='ica_aroma_confound_extraction')
 
+    ica_aroma_metadata_fmt = pe.Node(
+        TSV2JSON(index_column='IC', output=None, enforce_case=True,
+                 additional_metadata={'Method': {
+                                      'Name': 'ICA-AROMA',
+                                      'Version': '0.3-beta'}}),
```
Comment (oesteban) [inline fmriprep/workflows/bold/confounds.py]: Just to note that I've been looking into how to query ICA-AROMA's version, and it seems that hard-coding it is the only way. Please note that this metadata may not be true if the user has a bare-metal installation. Any thoughts @effigies? Perhaps remove it?

Comment (oesteban) [reply fmriprep/workflows/bold/confounds.py]: Okay, actually we are using v0.4.4-beta, as per Dockerfile.

Comment (effigies) [reply fmriprep/workflows/bold/confounds.py]: Would it make sense just to help package ICA-AROMA, make sure it has `__version__`, and put it on PyPI?

**Comment by oesteban** [comment]
At some point this evening I thought I'm mad when found myself with AROMA's
repo open and thinking about exactly that :p

A couple of things discouraged me from starting right away:

- First, the project seems a bit abandoned, judging by the outstanding PR
proposing a 0.4.4 release. We will need a fast turnaround
- The other is that we might just implement it as an interface for nipype
like compcor. The current implementation has a lot of boilerplate to run
normalization and MELODIC which we don't need. Maybe just picking up the
classification of components would suffice.

**Comment by oesteban** [comment]
Just to note: this PR is still failing (https://circleci.com/gh/poldracklab/fmriprep/9634). It doesn't look like a big deal.

```
Node: fmriprep_wf.single_subject_01_wf.func_preproc_task_mixedgamblestask_run_02_wf.ica_aroma_wf.ica_aroma_metadata_fmt
Working directory: /scratch/fmriprep_wf/single_subject_01_wf/func_preproc_task_mixedgamblestask_run_02_wf/ica_aroma_wf/ica_aroma_metadata_fmt

Node inputs:

additional_metadata = {'Method': {'Name': 'ICA-AROMA', 'Version': '0.4.4-beta'}}
drop_columns = None
enforce_case = True
in_file = /scratch/fmriprep_wf/single_subject_01_wf/func_preproc_task_mixedgamblestask_run_02_wf/ica_aroma_wf/ica_aroma_confound_extraction/classification_overview.tsv
index_column = IC
output = None

Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/legacymultiproc.py", line 69, in run_node
    result['result'] = node.run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 473, in run
    result = self._run_interface(execute=True)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 564, in _run_interface
    return self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 649, in _run_command
    result = self._interface.run(cwd=outdir)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 376, in run
    runtime = self._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.7/site-packages/niworkflows/interfaces/utils.py", line 822, in _run_interface
    enforce_case=self.inputs.enforce_case
  File "/usr/local/miniconda/lib/python3.7/site-packages/niworkflows/interfaces/utils.py", line 879, in _tsv2json
    tsv_data[k] = v
  File "/usr/local/miniconda/lib/python3.7/site-packages/pandas/core/frame.py", line 3119, in __setitem__
    self._set_item(key, value)
  File "/usr/local/miniconda/lib/python3.7/site-packages/pandas/core/frame.py", line 3194, in _set_item
    value = self._sanitize_column(key, value)
  File "/usr/local/miniconda/lib/python3.7/site-packages/pandas/core/frame.py", line 3391, in _sanitize_column
    value = _sanitize_index(value, self.index, copy=False)
  File "/usr/local/miniconda/lib/python3.7/site-packages/pandas/core/series.py", line 4001, in _sanitize_index
    raise ValueError('Length of values does not match length of ' 'index')
ValueError: Length of values does not match length of index
```

**Pull request merged**